pkg/sink/kafka/v2/kafka-go-client.go

```diff
@@ -0,0 +1,363 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"context"
+	"crypto/tls"
+	"strings"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/contextutil"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/pkg/errors"
+	pkafka "github.com/pingcap/tiflow/pkg/sink/kafka"
+	"github.com/pingcap/tiflow/pkg/util"
+	"github.com/rcrowley/go-metrics"
+	"github.com/segmentio/kafka-go"
+	"github.com/segmentio/kafka-go/sasl"
+	"github.com/segmentio/kafka-go/sasl/plain"
+	"github.com/segmentio/kafka-go/sasl/scram"
+	"github.com/segmentio/kafka-go/topics"
+	"go.uber.org/zap"
+)
+
+type kafkaGoClient struct {
+	transport *kafka.Transport
+	client    *kafka.Client
+	options   *pkafka.Options
+}
+
+// NewKafkaGoClient constructs a Client with kafka go.
+func NewKafkaGoClient(ctx context.Context, options *pkafka.Options) (pkafka.Client, error) {
+	var (
+		err       error
+		clientID  string
+		tlsConfig *tls.Config
+		mechanism sasl.Mechanism
+	)
+	if options.EnableTLS {
+		tlsConfig = &tls.Config{
+			MinVersion: tls.VersionTLS12,
+			NextProtos: []string{"h2", "http/1.1"},
+		}
+
+		// for SSL encryption with self-signed CA certificate, we reassign the
+		// config.Net.TLS.Config using the relevant credential files.
+		if options.Credential != nil && options.Credential.IsTLSEnabled() {
+			tlsConfig, err = options.Credential.ToTLSConfig()
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+	}
+	captureAddr := contextutil.CaptureAddrFromCtx(ctx)
+	changefeedID := contextutil.ChangefeedIDFromCtx(ctx)
+	var role string
+	if contextutil.IsOwnerFromCtx(ctx) {
+		role = util.RoleOwner.String()
+	} else {
+		role = util.RoleProcessor.String()
+	}
+	clientID, err = pkafka.NewKafkaClientID(role, captureAddr, changefeedID, options.ClientID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	mechanism, err = completeSASLConfig(options)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	transport := &kafka.Transport{
+		SASL:        mechanism,
+		ClientID:    clientID,
+		TLS:         tlsConfig,
+		IdleTimeout: options.DialTimeout,
+	}
+	client := &kafka.Client{
+		Addr:      kafka.TCP(options.BrokerEndpoints...),
+		Timeout:   10 * time.Second,
```

Comment:
it looks this timeout should be configurable, we can make it happen later.

Reply:
added todo